### DISCUSSION TITLE
x:bind not updating real time, x:text for same code updates

### DISCUSSION THREAD

**Discussion by muhaym** [alpinejs/alpine#1364] [1. Help]
Hi, 

I have two variables currency and billPlan, these are updated in real time based on a switch control on the page.
If I use 
` x-text="plan.name+'-'+currency+'-'+billPlan" `
and try to change the switch status, I can see real-time change wherever I'm using x-text.

But if use
` x-bind:data-cb-item-0="plan.name+'-'+currency+'-'+billPlan"`
and try to change the switch status, the changes are not reflected, only initial value is loaded. 

I'm pretty new to alpine.js. What might be the mistake I'm doing here?

**Comment by KevinBatdorf** [comment]
Can you share the rest of your code, in a demo like codepen if possible?

It usually depends on how you're updating it, but there is nuance as well so it depends.

**Reply by muhaym** [reply]
Hi

here is a small demo of what I'm trying to do.
https://codepen.io/muhaym/pen/YzZeoVX

**Reply by KevinBatdorf** [reply]
That seems to work fine for me. Can you provide step by step instruction how to break it?

**Reply by muhaym** [reply]
The change works for x-text, It doesn't work for x-bind, didn't get a quick way to show that (probably fire a click event and show the value of data-cb-item-0 attribute?)

**Reply by KevinBatdorf** [reply]
Yeah, but I click the button and all the bound attributes update properly. Did you try the demo?

![attrs](https://user-images.githubusercontent.com/1478421/120661631-a7995a00-c4b2-11eb-94a0-a1c73bd14711.gif)

Is there a specific attribute not updating? Is it happening in a specific browser?

**Reply by muhaym** [reply]
It's not working for me in both Chrome as well as Firefox (but I'm not able to debug as you did, in Inspect element what I see is the same code we wrote).

Whenever I click the button, in overall code a popup appears taking the value from data-cb-item-0 attribute, which is chargebee javascript, something for a no-code subscription management tool. If I hard code, all the plans works as expected, but with expected binding, only initial value works. You may try here https://onboard.takemyorder.io/index-test.html

**Reply by muhaym** [reply]
Ok found debugging shows bound value in Chrome Inspector. In my case, it's not changing like yours (in local)
![image](https://user-images.githubusercontent.com/10905479/120665028-0d80e400-c4a9-11eb-9fe9-d1590cde3262.png)

**Reply by KevinBatdorf** [reply]
Even your test page works for me. Can you try in incognito? You could have a browser extension causing issues. 

![annual](https://user-images.githubusercontent.com/1478421/120671737-dff16600-c4bb-11eb-9de5-0d5e23ed3659.gif)

Ask some other people to check too. Do you get any console errors?

**Reply by muhaym** [reply]
Ok. Thanks for this update. This is interesting. It worked now in Edge and Chrome (I mean the data-cb-item-0) change was displayed in the inspect element. 

But this change doesn't seem to pass to the Chargebee library (it just takes the default plan)
![image](https://user-images.githubusercontent.com/10905479/120673274-a7985a80-c4b0-11eb-8c3b-680d70ae9b39.png)


As from the image - Grow-INR-Yearly should show INR currency and Yearly plan (If I try this with vanilla HTML and JS without alpinejs) it works fine
![image](https://user-images.githubusercontent.com/10905479/120673548-e6c6ab80-c4b0-11eb-83fc-66a78cf10a65.png)

**Reply by KevinBatdorf** [reply]
I'd probably contact Chargbee and see how they process that. I've dealt with them before and they have fixed bugs in that modal window before and were pretty responsive.
